src/styles/components/_menu.scss

```diff
@@ -1,91 +1,147 @@
-$interface-types: (
-  'name-generator',
-  'custom'
-);
+@keyframes menu-button-move {
+  0%    { transform: translate3d(0, -250%, 0); }
+  25%   { transform: translate3d(0, -250%, 0); }
+  100%  { transform: translate3d(0, 0, 0); }
+}
 
 .menu {
-  // General sidebar styles
+  &__wrap {
+    // General sidebar styles
+    height: 100%;
+    left: 0;
+    position: fixed;
+    transform: translate3d(-100%, 0, 0);
+    transition: all $animation-standard-duration $animation-default-easing;
+    z-index: 2;
+
+    &--open {
+      transform: translate3d(0%, 0, 0);
+    }
+    @include breakpoint(medium) {
+      width: 30vw;
+    }
+
+    @include breakpoint(large) {
+      width: 22vw;
+    }
+  }
+
   &__content {
-    background: palette(menu);
+    background: palette('menu');
+    color: palette('menu-headings');
+    display: flex;
     height: 100%;
-    color: $charcoal;
   }
 
   // Menu header
   &__title {
-    padding: $padding-base-vertical $spacer-4;
-    color: palette(menu-headings);
+    color: palette('menu-headings');
     font-weight: 700;
     line-height: 2.5rem;
+    padding: $padding-base-vertical $spacer-4;
   }
 
   &__search {
     padding: 1rem;
 
     input {
+      border-radius: .5rem;
+      outline: none;
       padding: .3rem;
       width: 100%;
+
+      &:focus {
+        border-width: 3px;
+      }
+
+      &:not(:focus) {
+        margin: 1px;
+      }
     }
   }
 
   // Wrapper for item list
   &__menuitem {
     align-items: center;
-    border-top: 1px solid $divider;
-    color: $charcoal;
+    background-position: 25px 50%;
+    background-repeat: no-repeat;
+    background-size: 35px;
+    border-top: 1px solid palette('divider');
+    color: palette('menu-headings');
+    cursor: pointer;
     display: flex;
     font-weight: 500;
     letter-spacing: 3px;
-    padding: 15px 25px 15px 80px;
+    padding: 15px 25px 15px 20px;
     text-decoration: none;
     text-transform: uppercase;
-    background-size: 35px;
-    background-repeat: no-repeat;
-    background-position: 25px 50%;
-    cursor: pointer;
+
+    &--settings:not(:hover):not(.menu__menuitem--active) .icon {
```

Comment:
Do we need this extra selectors? If the "non-active" state is this colour, can we not just have `&--settings .icon`?